src/learning/nnet.rs

```diff
@@ -328,10 +311,9 @@ impl<'a, T: Criterion> BaseNeuralNet<'a, T> {
             }
         }
 
-        let mut grad = Vec::with_capacity(self.layer_sizes.len() - 1);
-        let mut capacity = 0;
+        let mut gradients = Vec::new();
```

Comment:
I'm new in machine learning, but I'd say we'll have as many `gradients` as we have input `weights` ?